packages/http-framework/src/lib/Client.ts

```diff
@@ -52,29 +54,68 @@ export class Client extends EventEmitter {
 	 */
 	public async listen({ serverOptions, postPath, port, address, ...listenOptions }: ListenOptions) {
 		const key = await makeKey(this.#discordPublicKey);
+		const path = postPath ?? process.env.HTTP_POST_PATH ?? '/';
 
-		this.server = Fastify(serverOptions);
-		this.server.addHook('onRoute', (options) => assignRawData(options));
-		this.server.addHook('preHandler', (request, reply) => handleSecurityHook(request, reply, key));
-		this.server.post(postPath ?? process.env.HTTP_POST_PATH ?? '/', this.handleHttpMessage.bind(this));
+		this.server = createServer(serverOptions ?? {});
+		this.server.on('request', (request, response) => this.handleRawHttpMessage(request, response, path, key));
 
-		await this.server.listen({ ...listenOptions, port, host: address });
+		return new Promise<void>((resolve) => this.server.listen({ ...listenOptions, port, host: address }, resolve));
 	}
 
-	protected async handleHttpMessage(request: FastifyRequest, reply: FastifyReply): Promise<FastifyReply> {
-		const interaction = request.body as APIInteraction;
-		if (interaction.type === InteractionType.Ping) return reply.send({ type: InteractionResponseType.Pong });
+	protected async handleRawHttpMessage(request: IncomingMessage, response: ServerResponse, path: string, key: Key) {
+		response.setHeader('Content-Type', 'application/json');
+
+		if (request.url !== path) {
+			response.statusCode = HttpCodes.NotFound;
+			return response.end('{"message":"Unsupported method"}');
+		}
+
+		if (request.method !== 'POST') {
+			response.statusCode = HttpCodes.MethodNotAllowed;
+			return response.end('{"message":"Unsupported method"}');
+		}
+
+		const signature = request.headers['x-signature-ed25519'];
+		const timestamp = request.headers['x-signature-timestamp'];
+
+		if (isNullishOrEmpty(signature) || isNullishOrEmpty(timestamp)) {
+			response.statusCode = HttpCodes.Unauthorized;
+			return response.end('{"message":"Missing signature information"}');
+		}
+
+		const body = await text(request);
+		if (isNullishOrEmpty(body)) {
+			response.statusCode = HttpCodes.BadRequest;
+			return response.end('{"message":"Missing body data"}');
+		}
+
+		const valid = await verifyBody(body, signature, timestamp, key);
+		if (!valid) {
+			response.statusCode = HttpCodes.Unauthorized;
+			return response.end('{"message":"The signature is incorrect"}');
+		}
+
+		return this.handleHttpMessage(JSON.parse(body) as APIInteraction, response);
+	}
+
+	protected async handleHttpMessage(interaction: APIInteraction, response: ServerResponse): Promise<ServerResponse> {
+		if (interaction.type === InteractionType.Ping) {
+			response.statusCode = HttpCodes.OK;
+			return response.end(`{"type":${InteractionResponseType.Pong}}`);
```

Comment:
Please don't use json strings, but JSON.stringified objects. Last thing you need is debugging a missing quote